src/huggingface_hub/cli/repo.py

```diff
@@ -160,6 +166,114 @@ def repo_settings(
     print(f"Successfully updated the settings of {ANSI.bold(repo_id)} on the Hub.")
 
 
+@repo_cli.command("list", help="List repositories (models, datasets, spaces) hosted on the Hub.")
+def repo_list(
+    repo_type: RepoTypeOpt = RepoType.model,
+    limit: Annotated[
+        int,
+        typer.Option(help="Limit the number of results."),
+    ] = 10,
+    filter: Annotated[
+        Optional[list[str]],
+        typer.Option(help="Filter by tags (e.g. 'text-classification'). Can be used multiple times."),
+    ] = None,
+    search: Annotated[Optional[str], typer.Option(help="Search by name.")] = None,
+    author: Annotated[Optional[str], typer.Option(help="Filter by author or organization.")] = None,
+    sort: Annotated[
+        Optional[str],
+        typer.Option(help="Sort results key, optionally with direction (e.g. 'likes', 'downloads:asc')."),
+    ] = None,
+    token: TokenOpt = None,
+) -> None:
+    """List repositories of the requested type and print them as JSON."""
+    api = get_hf_api(token=token)
+
+    sort_key: Optional[str] = None
+    direction: Optional[Literal[-1, 1]] = None
+
+    if sort:
+        match = _SORT_PATTERN.match(sort)
+        if not match:
+            typer.echo(
+                f"Error: Invalid sort format '{sort}'. Expected 'field' or 'field:direction' (e.g. 'downloads:desc')."
+            )
+            raise typer.Exit(1)
+
+        sort_key = match.group("key")
+        order = match.group("order")
+
+        if order == "asc":
+            direction = 1
+        elif order == "desc":
+            direction = -1
```

Comment:
No need to set `direction=1` for `asc`, `None` means ascending
```suggestion
        if order == "desc":
           direction = -1
```